app/src/trezor.js

```diff
@@ -0,0 +1,286 @@
+/* eslint-disable no-bitwise */
+/* eslint-disable */
+
+import { ipcMain } from 'electron'; // eslint-disable-line import/no-extraneous-dependencies
+import win from './modules/win';
+import {
+  HWDevice,
+  addConnectedDevices,
+  removeConnectedDeviceByID,
+  getDeviceById } from './hwManager';
+
+const trezor = require('trezor.js'); // eslint-disable-line import/no-extraneous-dependencies
+
+// set to true to see messages
+const debug = false;
+const logDebug = (...args) => {
+  if (debug) {
+    // eslint-disable-next-line no-console
+    console.log(...args);
+  }
+};
+
+// DeviceList encapsulates transports, sessions, device enumeration and other
+// low-level things, and provides easy-to-use event interface.
+const list = new trezor.DeviceList({ debug });
+
+const createTrezorHWDevice = deviceFeatures =>
+  new HWDevice(
+    deviceFeatures.device_id,
+    deviceFeatures.label,
+    (deviceFeatures.model === '1' ? 'Trezor One' : 'Trezor Model T'),
+    null,
+  );
+
+const hardeningConstant = 0x80000000;
+const getHardenedPath = index => [
+  (44 | hardeningConstant) >>> 0,
+  (134 | hardeningConstant) >>> 0,
+  (index | hardeningConstant) >>> 0,
+];
+
+/**
+ * @param {string} label: Label of Trezor Device
+ * @param {string} code: Codes are in the format ButtonRequest_[type]
+ * where [type] is one of the types, defined here:
+ * https://github.com/trezor/trezor-common/blob/master/protob/types.proto#L78-L89
+ */
+const buttonCallback = (label) => {
+  logDebug('buttonCallback:', label);
+  win.send({ event: 'trezorButtonCallback', value: label });
+}; 
+
+/**
+ * @param {string} type
+ * @param {Function<Error, string>} callback
+ */
+const pinCallback = (type, callback) => {
+  logDebug('pinCallback:', type);
+  win.send({ event: 'trezorPinCallback', value: null });
+  ipcMain.once('trezorPinCallbackResponse', (event, pin) => {
+    if (pin) {
+      callback(null, pin);
+    } else {
+      callback('pin_not_provided_from_ui', null);
+    }
+  });
+};
+
+/**
+ * @param {string} deviceId of Trezor
+ * @param {Function<Error, string>} callback
+ */
+const passphraseCallback = (deviceId, callback) => {
+  logDebug('passphraseCallback:', deviceId);
+  // We check if we have saved it (Trezor One only)
+  const td = getDeviceById(deviceId);
+  if (td && td.pp) {
+    callback(null, td.pp);
+    return;
+  }
+
+  win.send({ event: 'trezorPassphraseCallback', value: null });
+
+  ipcMain.once('trezorPassphraseCallbackResponse', (event, passphrase) => {
+    if (passphrase || passphrase === '') {
+      callback(null, passphrase);
+      // Save passphrase in case of Trezor One (it asks on every action. Pretty annoying)
+      const trezorDevice = getDeviceById(deviceId);
+      if (trezorDevice && trezorDevice.model === 'Trezor One') {
```

Comment:
I think is better define all information related to HW in a constants file, then in the future if we want to update something will be only in the constants file